src/xccl/ProcessGroupXCCL.cpp

```diff
@@ -391,8 +429,13 @@ std::shared_ptr<xcclComm_t> ProcessGroupXCCL::getXCCLComm(
   }
 
   c10::impl::VirtualGuardImpl impl(device.type());
-  c10::Stream stream =
-      impl.getStreamFromGlobalPool(device, /*isHighPriority=*/false);
+
+  // asyncOp=false will always use current stream; getStrem will return current
+  // stream
+  c10::Stream stream = asyncOp
```

Comment:
In some special case like FSDP2, same communicator will have different `current stream` on each device. Then how to make sure to return correct compute stream for communication?